namer/serversets/src/main/scala/io/buoyant/namer/serversets/ServersetNamer.scala

```diff
@@ -49,7 +54,12 @@ class ServersetNamer(zkHost: String, idPrefix: Path) extends Namer {
         val addr = if (segments.nonEmpty && (segments.last contains ":")) {
           val Array(name, endpoint) = segments.last.split(":", 2)
           val zkPath = (segments.init :+ name).mkString("/", "/", "")
-          resolveServerset(zkHost, zkPath, endpoint)
+          if (!(endpoint contains "#")) {
```

Comment:
By putting this inside the `if` on line 54, we're only able to use shard ids when also specifying an endpoint.  According to https://github.com/twitter/finagle/blob/develop/finagle-serversets/src/main/scala/com/twitter/finagle/serverset2/naming/ServersetPath.scala#L23 it looks like they should be able to be used independently.

Ideally here we would just use `ServersetPath`'s unapply to ensure that we parse this correctly.  Unfortunately, `ServersetPath` is marked as package private to `com.twitter`.  We get around that restriction elsewhere in Linkerd by moving classes to packages like `com.twitter.buoyant.namer.serversets`.  If you moved this class into that package, we could use `ServersetPath` and wouldn't need to worry about maintaining our own parsing code.

Reply:
Good find, I was wondering if that worked w/o the labeled part of the endpoint.

I'd be happy to refactor this part to use the `ServersetPath`'s unapply.